src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/basicshortmutpileup/BasicValidationResult.java

```diff
@@ -104,4 +118,112 @@ public String getFilters() {
     public long getNumAltSupportingReadsInNormal() {
         return numAltSupportingReadsInNormal;
     }
+
+    //----- The following two public static methods read and write files
+    public static void writeToFile(final List<BasicValidationResult> records, final File outputTable) {
+        try ( final BasicValidationResult.BasicValidationResultTableWriter writer = new BasicValidationResult.BasicValidationResultTableWriter(IOUtils.fileToPath(outputTable)) ) {
+            writer.writeHeaderIfApplies();
+            writer.writeAllRecords(records);
+        } catch (IOException e){
+            throw new UserException(String.format("Encountered an IO exception while writing to %s.", outputTable));
+        }
+    }
+
+    public static List<BasicValidationResult> readFromFile(final File tableFile) {
+        try( final BasicValidationResult.BasicValidationResultTableReader reader = new BasicValidationResult.BasicValidationResultTableReader(IOUtils.fileToPath(tableFile)) ) {
+            return reader.toList();
+        } catch (IOException e){
+            throw new UserException(String.format("Encountered an IO exception while reading from %s.", tableFile));
+        }
+    }
+
+    //-------- The following methods are boilerplate for reading and writing tables
+    private static class BasicValidationResultTableWriter extends TableWriter<BasicValidationResult> {
```

Comment:
Reconsider the access modifiers of the below classes (and their constructors) and enum (and its `COLUMNS` field)---some inconsistencies here.

Reply:
done